### PR TITLE
[SPARK-41803][CONNECT][PYTHON] Add missing function `log(arg1, arg2)`

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Add missing function `log(arg1, arg2)`

### Why are the changes needed?
for API coverage

### Does this PR introduce _any_ user-facing change?
yes

### How was this patch tested?
added ut

### CODE REVIEW

File: python/pyspark/sql/connect/functions.py [hunk 1]
```diff
@@ -551,6 +551,24 @@ def log(col: "ColumnOrName") -> Column:
     return _invoke_function_over_columns("ln", col)
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/functions.py:551]: shall we remove this?

File: python/pyspark/sql/connect/functions.py [hunk 2]
```diff
@@ -551,6 +551,24 @@ def log(col: "ColumnOrName") -> Column:
...
+def log(arg1: "ColumnOrName") -> Column:
+    ...
+
+
+@overload
+def log(arg1: float, arg2: "ColumnOrName") -> Column:
+    ...
+
+
+def log(arg1: Union["ColumnOrName", float], arg2: Optional["ColumnOrName"] = None) -> Column:
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/functions.py:564]: and add the default value here.

**Comment by zhengruifeng** [comment]
@HyukjinKwon  this PR just follows the existing design in PySpark, but I also feel it a bit problematic, the `log` function actually support only two cases:

1, `log(column)`;
2, `log(float_value, column)`

`log(column, column)` was actually not supported (both in PySpark and Scala)

```
py4j.Py4JException: Method log([class java.lang.String, class org.apache.spark.sql.Column]) does not exist
        at py4j.reflection.ReflectionEngine.getMethod(ReflectionEngine.java:321)
        at py4j.reflection.ReflectionEngine.getMethod(ReflectionEngine.java:342)
        at py4j.Gateway.invoke(Gateway.java:276)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
```

I will try to update the function signature

File: python/pyspark/sql/connect/functions.py (hunk 1 above)
Comment (HyukjinKwon) [reply python/pyspark/sql/connect/functions.py:551]: (Python doesn't support overload so this function definition will simply be overwritten and no-op)

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**